### PR TITLE
devices: stm32l4x6: add ram storage driver based on flash-stm32

### PR DESCRIPTION
**Pull request by damianloew**
## Description

 - devices: stm32l4x6: add ram storage driver based on flash-stm32
 - stm32l4x6: move FLASH defines from hal to linker template
 
## Motivation and Context

- It's needed for loading apps, previously placed in RAM memory (for example by gdb and jtag). Required for stm32l4x6-nucleo testing process.

- JIRA: PD-301

- Memory areas should be located in linker script
    
- JIRA: RTOS-245

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (stm32l4x6-nucleo).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Review by HBuczynski — APPROVED** [review]
LGTM

File: hal/armv7m/stm32/l4/peripherals.h [hunk 1]
```diff
@@ -73,4 +73,7 @@
 #define FLASH_PROGRAM_2_ADDR    0x08080000
 #define FLASH_PROGRAM_BANK_SIZE (512 * 1024)
 
+#define RAM_ADDR      0x20000000
+#define RAM_BANK_SIZE (320 * 1024)
```
Comment (gerard5) [inline hal/armv7m/stm32/l4/peripherals.h:77]: We have place for statically defined memory areas: [armv7m4-stm32l4x6.ldt](https://github.com/phoenix-rtos/plo/blob/master/ld/armv7m4-stm32l4x6.ldt).  Every statically defined memory area/region should be placed there.

Btw. not the case for this PR, but `FLASH_PROGRAM_*` should also be moved to the `*ldt` file.

Comment (damianloew) [reply hal/armv7m/stm32/l4/peripherals.h]: Moved in the last force-push, did You mean sth like that?

Comment (HBuczynski) [reply hal/armv7m/stm32/l4/peripherals.h]: I agree with @gerard5 that memory areas should be located in linker script.
I know that moving `FLASH_PROGRAM_*` define is not in the scope of this PR but you can do it in a separate commit within this PR.

Comment (damianloew) [reply hal/armv7m/stm32/l4/peripherals.h]: Done in second commit.

Comment (gerard5) [reply hal/armv7m/stm32/l4/peripherals.h]: Yes.

**Review by gerard5 — APPROVED** [review]
Tested on HW, works as expected.

**Pull request merged**